frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -599,8 +594,12 @@ final class BloopBspServices(
         projects.iterator.map {
           case (target, project) =>
             val sources = project.sources.map { s =>
+              val uri = s.toBspSourceUri
```

Comment:
We could also do it like:
```
                val kind = if (s.exists) {
                  if (s.isFile) bsp.SourceItemKind.File else bsp.SourceItemKind.Directory
                } else {
                  if (AbsolutePath.sourceFileNameMatcher.matches(s.underlying.getFileName))
                    bsp.SourceItemKind.File
                  else bsp.SourceItemKind.Directory
                }
```

But whatever you think is best.

Reply:
Didn't go for this approach because it's basically inlining `toBspSourceUri`, but given that we only use `toBspSourceUri` once in all the codebase (we used to use it more) the most sensible thing might be to just inline it here and remove `toBspSourceUri` altogether. Can you push  that change :smile: ? 
  
I'll investigate what's up with Windows later on the day

Reply:
Sure, I will take care of it :)